src/ui/public/courier/data_source/build_query/from_kuery.js

```diff
@@ -0,0 +1,52 @@
+import _ from 'lodash';
+import { fromKueryExpression } from '../../../kuery';
+
+function astToQuery(ast) {
+  if (Array.isArray(ast)) {
```

Comment:
Agreed, I was being a bit lazy here. Why are you thinking the type would be `or`? Some children might be required, while others might be optional. Could you elaborate on how `left` `right` would work?